include/mxnet/c_api.h

```diff
@@ -227,10 +227,138 @@ MXNET_DLL int MXSetProfilerConfig(int mode, const char* filename);
  */
 MXNET_DLL int MXSetProfilerState(int state);
 
-/*! \brief Save profile and stop profiler */
+/*!
+ * \brief Save profile and stop profiler
+ * \param append true if appending to current profile file, false for truncate
+ * \return
+ */
 MXNET_DLL int MXDumpProfile();
 
-/*! \brief Set the number of OMP threads to use */
+/*!
+ * \brief Set whether to continuously write the profiling data to a file
+ * \param continuous_dump true to continuously write profiling data to a file
+ * \param delay_in_seconds Number of seconds (or fraction of seconds) to delay between writes
+ * \return 0 when success, -1 when failure happens.
+ */
+MXNET_DLL int MXSetContinuousProfileDump(int continuous_dump, float delay_in_seconds);
+
+/*!
+ * \brief Control whether to append profile data dumps to the output file or whether to truncate
+ * \param append true to append subsequent MXDumpProfile() calls to the preexisting file
+ * \return 0 when success, -1 when failure happens.
+ */
+MXNET_DLL int MXSetDumpProfileAppendMode(int append);
```

Comment:
Why are these part of config?

Reply:
Why not? I’d like to include them as part of the normal Config call, but am wary about changing the API which you usually seem to dislike

Reply:
I have considered removing the append mode, however. It’s sort of implied by the continuous dump, but was trying to make the old behavior still possible with the right settings.

Reply:
Why not merge this with the config api?

Reply:
Ok

Reply:
Done